services/streammanager/bqstream/bqstreammanager.go

```diff
@@ -70,12 +80,18 @@ func NewProducer(destinationConfig interface{}) (*bigquery.Client, error) {
 			gbq.BigqueryInsertdataScope,
 		}...),
 	}
-	return bigquery.NewClient(context.Background(), config.ProjectId, opts...)
+	bqClient, err := bigquery.NewClient(context.Background(), config.ProjectId, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{bqClient: bqClient, o: o}, nil
 }
 
 func Produce(jsonData json.RawMessage, producer interface{}, destConfig interface{}) (statusCode int, respStatus string, responseMessage string) {
 
-	bqClient := producer.(*bigquery.Client)
+	client := producer.(*Client)
```

Comment:
[non-blocking] This part of the code could use some refactoring. 

It is really strange that we use a function instead of a method, and we are passing the client as an interface{}.

I would expect to see something like this:
```go
func (c *Client) Produce(jsonData json.RawMessage, destConfig interface{}) {

```

I understand that this is beyond this PR and the task we have picked for this sprint. I've open a ticket for this https://www.notion.so/rudderstacks/Test-refactor-streammanager-package-fedb6738d79e4fe9b5e027989da98e48